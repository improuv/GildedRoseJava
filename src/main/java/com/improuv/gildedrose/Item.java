package com.improuv.gildedrose;

/**
 * DO NOT TOUCH --
 * However, do not alter the Item class or Items property as those belong to the goblin in the 
 * corner who will insta-rage and one-shot you as he doesn't believe in shared code ownership
 */
public class Item {
    private final String name;
    private int sellIn;
    private int quality; 
    
    public Item(String name, int sellIn, int quality) {
        this.name = name;
        this.setSellIn(sellIn);
        this.setQuality(quality);
    }
    
    public String getName() {
        return name;
    }
    public int getSellIn() {
        return sellIn;
    }
    public void setSellIn(int sellIn) {
        this.sellIn = sellIn;
    }
    public int getQuality() {
        return quality;
    }
    public void setQuality(int quality) {
        this.quality = quality;
    }
    
    @Override
    public String toString() {
    	return getName() + "[sellin=" + getSellIn() + " quality=" + getQuality() + "]";
    }
}
