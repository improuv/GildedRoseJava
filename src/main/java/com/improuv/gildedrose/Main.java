package com.improuv.gildedrose;
import java.util.ArrayList;

public class Main {

	/**
	 * @param args
	 */
	public static void main(String[] args) {

		System.out.println("OMGHAI!");

		GildedRose app = new GildedRose();

		app.items = new ArrayList<Item>();
		app.items.add(new Item("+5 Dexterity Vest", 10, 20));
		app.items.add(new Item("Aged Brie", 2, 0));
		app.items.add(new Item("Elixir of the Mongoose", 5, 7));
		app.items.add(new Item("Sulfuras, Hand of Ragnaros", 0, 80));
		app.items.add(new Item("Backstage passes to a TAFKAL80ETC concert", 15,
				20));
		app.items.add(new Item("Conjured Mana Cake", 3, 6));

		app.updateQuality();
	}

}
