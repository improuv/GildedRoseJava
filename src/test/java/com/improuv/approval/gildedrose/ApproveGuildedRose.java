package com.improuv.approval.gildedrose;

import org.approvaltests.Approvals;
import org.junit.Test;

import com.improuv.gildedrose.GildedRose;
import com.improuv.gildedrose.Item;

public class ApproveGuildedRose {

    private static final String BACKSTAGE_PASS_NAME = "Backstage passes to a TAFKAL80ETC concert";
    private static final String AGED_BRIE_NAME = "Aged Brie";
    public static final String SULFURAS_NAME = "Sulfuras, Hand of Ragnaros";

    private static final int MIN_QUALITY = 0;
    private static final int MAX_QUALITY = 50;

    //@Test 
    public void approve()
    {
        final Item[] items = new Item[] {
                new Item("Item", 100, 1),
                new Item("Item", 0, 10),
                new Item("Item", 1000, MIN_QUALITY),
                new Item("Item", 14, MAX_QUALITY),
                new Item("Item", 0, MIN_QUALITY),
                new Item(SULFURAS_NAME, 5, 80),
                new Item(SULFURAS_NAME, -1, 80),
                new Item(AGED_BRIE_NAME, 5, 15),
                new Item(AGED_BRIE_NAME, 0, 15),
                new Item(AGED_BRIE_NAME, 5, MAX_QUALITY),
                new Item(AGED_BRIE_NAME, 0, MAX_QUALITY),
                new Item(BACKSTAGE_PASS_NAME, 11, 30),
                new Item(BACKSTAGE_PASS_NAME, 10, 30),
                new Item(BACKSTAGE_PASS_NAME, 6, 30),
                new Item(BACKSTAGE_PASS_NAME, 5, 30),
                new Item(BACKSTAGE_PASS_NAME, 0, 30),
                new Item(BACKSTAGE_PASS_NAME, 5, 50),
                new Item(BACKSTAGE_PASS_NAME, 5, 49),
                new Item(BACKSTAGE_PASS_NAME, -1, 10),

        };
        final GildedRose gildedRose = GildedRose.with(items);
        gildedRose.updateQuality();
        Approvals.verifyAll("", items);
    }
}
