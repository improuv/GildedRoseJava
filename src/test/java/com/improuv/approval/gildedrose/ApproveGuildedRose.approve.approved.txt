[00] = Item[sellin=99 quality=0]
[01] = Item[sellin=-1 quality=8]
[02] = Item[sellin=999 quality=0]
[03] = Item[sellin=13 quality=49]
[04] = Item[sellin=-1 quality=0]
[05] = Sulfuras, Hand of Ragnaros[sellin=5 quality=80]
[06] = Sulfuras, Hand of Ragnaros[sellin=-1 quality=80]
[07] = Aged Brie[sellin=4 quality=16]
[08] = Aged Brie[sellin=-1 quality=17]
[09] = Aged Brie[sellin=4 quality=50]
[10] = Aged Brie[sellin=-1 quality=50]
[11] = Backstage passes to a TAFKAL80ETC concert[sellin=10 quality=31]
[12] = Backstage passes to a TAFKAL80ETC concert[sellin=9 quality=32]
[13] = Backstage passes to a TAFKAL80ETC concert[sellin=5 quality=32]
[14] = Backstage passes to a TAFKAL80ETC concert[sellin=4 quality=33]
[15] = Backstage passes to a TAFKAL80ETC concert[sellin=-1 quality=0]
[16] = Backstage passes to a TAFKAL80ETC concert[sellin=4 quality=50]
[17] = Backstage passes to a TAFKAL80ETC concert[sellin=4 quality=50]
[18] = Backstage passes to a TAFKAL80ETC concert[sellin=-2 quality=0]
